### DISCUSSION TITLE
Getting a user with the simplest Express server has problems

### DISCUSSION THREAD

**Discussion by elanzini** [supabase/supabase#9764] [Questions] [answered]
Hello,
I created the simplest express server I could and a `post` route where the client is sending the user `access_token` and the idea is to create a new client and use the `jwt` to get the user and the id of the user to retrieve more information from other tables.

What I am trying to achieve is similar to [this](https://github.com/supabase/supabase/discussions/1094).
The code is based on [this documentation](https://supabase.com/docs/reference/javascript/v1/auth-api-getuser) of the `getUser()`

Here is the code:
```javascript
require("dotenv").config();

const express = require("express");
const cors = require("cors");
const { createClient } = require("@supabase/supabase-js");
const bodyParser = require("body-parser");

const app = express();
const port = 8080;

const supabaseUrl = process.env.SUPABASE_URL;
const supabaseAnonKey = process.env.SUPABASE_ANON_KEY;

app.use(bodyParser.urlencoded({ extended: true }));
app.use(bodyParser.json());

app.use(cors());

app.post("/", async (req, res) => {
  const jwt = req.body.supabase_access_token;
  const supabase = createClient(supabaseUrl, supabaseAnonKey);
  const { user, error } = await supabase.auth.api.getUser(jwt); // <-- ERROR

  res.sendStatus(200);
});

app.listen(port, () => {
  console.log(`Server listening on port ${port}`);
});
```

The error is:
```
const { user, error } = await supabase.auth.api.getUser(jwt);
TypeError: Cannot read properties of undefined (reading 'getUser')
```

Am I doing something incorrectly?

**Comment by GaryAustin1** [comment, marked as answer]
That is Supabase v1 format.  Are you maybe using Supabase-js v2 client code?  If so the format is here https://supabase.com/docs/reference/javascript/auth-getuser

**Reply by elanzini** [reply]
That was it - sorry for the confusion

**Comment by Cancuuu** [comment]
Sorry guys, but is it a good practice create a Supabase client `createClient()` for each request? just that. Thank you. 💚

**Reply by GaryAustin1** [reply]
That is normal practice on serverside code.  You don't want the client session data being shared.
createClient normally only makes a Supabase server call if the user access token has expired.
